### PR TITLE
Add BIT support

### PR DESCRIPTION
**Pull request by JelteF**
This adds support for reading/writing the BIT type. Since there is no native type for representing bitstrings in Go, this adds a custom Go type that represents one. It uses the same internal storage as DuckDB so that conversions are cheap and require minimal copying.

P.S. Strictly speaking there is the BitString type in the encoding/asn1, but that's not meant for general purpose use. It's really only there so that X509 certificates can be decoded. It's also laid out very differently than DuckDB its BIT type (e.g. it uses 0's for padding instead of 1's).

### CODE REVIEW

File: types.go [hunk 1]
```diff
@@ -137,6 +138,103 @@ func uuidToHugeInt(uuid UUID) mapping.HugeInt {
...
+// Bit represents a DuckDB BIT value as a sequence of bits.
+// Internally, data stores DuckDB's internal format: a padding-count prefix byte followed by
+// the bit bytes (right-aligned with 1-padded MSB bits).
+// For example, "10101" (5 bits) is stored as [3, 11110101] where 3 is the padding count.
+type Bit struct {
+	data []byte
+}
+
+// NewBitFromString creates a Bit from a string of '0' and '1' characters.
+func NewBitFromString(s string) (*Bit, error) {
```
Comment (mlafeldt) [inline types.go]: This returns `*Bit` while all other type constructors return values - should be `Bit, error)` for consistency

File: value.go [hunk 2]
```diff
@@ -61,6 +61,10 @@ func getValue(v mapping.Value) (any, error) {
 	case TYPE_INTERVAL:
 		interval := mapping.GetInterval(v)
 		return getInterval(&interval), nil
+	case TYPE_BIT:
+		bit := mapping.GetBit(v)
+		defer mapping.DestroyBit(&bit)
+		return Bit{data: mapping.BitMembers(&bit)}, nil
```
Comment (mlafeldt) [inline value.go]: In this file, `TYPE_BIT` is still marked as "Invalid or unsupported" in `isPrimitiveType()` and missing from `inferPrimitiveType()`

File: types.go [hunk 3]
```diff
@@ -137,6 +138,103 @@ func uuidToHugeInt(uuid UUID) mapping.HugeInt {
...
+
+// NewBitFromRaw creates a Bit from bytes in DuckDB's internal format.
+// The first byte is the padding count (0-7), followed by bit data bytes that are right-aligned
+// with 1-padded MSB bits.
+// For example, NewBitFromRaw([]byte{6, 0xFE, 0xAB}) creates "1010101011":
+// padding=6, then [11111110 10101011] where the high 6 bits (111111) are padding.
+// The slice is consumed directly without copying, so the caller must not modify it afterwards.
+func NewBitFromRaw(data []byte) (*Bit, error) {
+	if len(data) <= 1 {
+		return &Bit{}, nil
```
Comment (mlafeldt) [inline types.go]: Nitpicky inconsistency:
 
- NewBitFromString("") → error ("empty bit string")
- NewBitFromRaw(nil) → success, returns &Bit{}

File: appender_test.go [hunk 4]
```diff
@@ -818,6 +793,57 @@ func TestAppenderTimeTZ(t *testing.T) {
 	require.Equal(t, expected, r)
 }
 
+func TestAppenderBit(t *testing.T) {
```
Comment (mlafeldt) [inline appender_test.go:845]: Can we also test scanning non-null into `var r *Bit`?

**Review by mlafeldt — COMMENTED** [review]
Thanks! Found some minor inconsistencies and missing tests. Otherwise looks great.

File: types.go [hunk 5]
```diff
@@ -136,6 +137,97 @@ func uuidToHugeInt(uuid UUID) mapping.HugeInt {
...
+		}
+	}
+
+	return Bit{Data: data}, nil
+}
+
+// Validate checks that Data is a valid DuckDB bit encoding: the padding count
+// (first byte) must be 0-7, and the padding bits in the first data byte must
+// all be set to 1.
+func (b Bit) Validate() error {
```
Comment (mlafeldt) [inline types.go:221]: This is never called on any write path. Both `setBit` and `bindBit` pass `Data` (which is public) straight to DuckDB without validation.

File: types.go [hunk 6]
```diff
@@ -136,6 +137,97 @@ func uuidToHugeInt(uuid UUID) mapping.HugeInt {
...
+	if len(b.Data) <= 1 {
+		return nil
```
Comment (mlafeldt) [inline types.go]: We also need to reject single-byte with nonzero padding

```golang
if len(b.Data) == 1 && b.Data[0] != 0 {
    return fmt.Errorf("invalid padding count %d with no data bytes", b.Data[0])
}
```

File: types.go [hunk 7]
```diff
@@ -136,6 +137,97 @@ func uuidToHugeInt(uuid UUID) mapping.HugeInt {
...
+// the bit bytes (right-aligned with 1-padded MSB bits).
+// For example, "10101" (5 bits) is stored as [3, 11110101] where 3 is the padding count.
+type Bit struct {
+	Data []byte
+}
+
+// NewBitFromString creates a Bit from a string of '0' and '1' characters.
+func NewBitFromString(s string) (Bit, error) {
+	if len(s) == 0 {
+		return Bit{}, nil
```
Comment (mlafeldt) [inline types.go]: This will return `Bit{Data: nil}`. Passing it to `setBit` or `bindBit` sends a nil slice into C code.

I suggest to either return an error for empty strings or return `Bit{Data: []byte{0}}` as the canonical empty encoding.

File: connection.go [hunk 8]
```diff
@@ -37,7 +37,7 @@ func newConn(conn mapping.Connection, ctxStore *contextStore) *Conn {
 // CheckNamedValue implements the driver.NamedValueChecker interface.
 func (conn *Conn) CheckNamedValue(nv *driver.NamedValue) error {
 	switch nv.Value.(type) {
-	case *big.Int, Interval, []any, []bool, []int8, []int16, []int32, []int64, []int, []uint8, []uint16,
+	case *big.Int, Interval, Bit, []any, []bool, []int8, []int16, []int32, []int64, []int, []uint8, []uint16,
```
Comment (mlafeldt) [inline connection.go]: This should also handle `*Bit`

File: statement.go [hunk 9]
```diff
@@ -435,6 +444,8 @@ func (s *Stmt) bindValue(val driver.NamedValue, n int) (mapping.State, error) {
 			return mapping.StateError, inferErr
 		}
 		return mapping.BindInterval(*s.preparedStmt, mapping.IdxT(n+1), i), nil
+	case Bit:
+		return s.bindBit(&v, n)
```
Comment (mlafeldt) [inline statement.go:451]: This should also handle `*Bit`

File: value.go [hunk 10]
```diff
@@ -372,7 +376,7 @@ func isPrimitiveType(t Type) bool {
 	case TYPE_DECIMAL, TYPE_ENUM, TYPE_LIST, TYPE_STRUCT, TYPE_MAP, TYPE_ARRAY, TYPE_UNION:
 		// Complex type.
 		return false
-	case TYPE_INVALID, TYPE_BIT, TYPE_ANY:
+	case TYPE_INVALID, TYPE_ANY:
```
Comment (mlafeldt) [inline value.go:388]: In `value.go`, `Bit` is still not handled in `inferPrimitiveType` and `createPrimitiveValue`

**Review by mlafeldt — CHANGES_REQUESTED** [review]
Thanks for the changes. I had another look and left a few more inline comments to address before merging.

**Comment by JelteF** [comment]
Okay, it took a while, but I addressed the review.

File: types.go (hunk 7 above)
Comment (JelteF) [reply types.go]: I've started erroring for empty bit strings. DuckDB does not allow them either.

**Review by mlafeldt — APPROVED** [review]
Thanks! Looks great now.

Two things I did on top of your work:

- I addressed a small appender *Bit asymmetry
- I made Bit scan-compatible
